src/controllers/auth.js

```diff
@@ -19,39 +23,50 @@ const register = async (req, res) => {
 
   try {
     const usedUsername = await UserModel.findOne({ username });
-    if (!usedUsername) {
-      const usedEmail = await UserModel.findOne({ email });
-      if (!usedEmail) {
-        const passwordHash = await bcrypt.hash(password0, 10);
-
-        await UserModel.create({
-          firstName,
-          lastName,
-          email,
-          username,
-          phoneNumber,
-          age,
-          gender,
-          nationality,
-          refugee,
-          provider: 'Local',
-          providerId: 'Local',
-          passwordHash,
-        });
-
-        return res.status(200).redirect('/api/auth/login');
-      }
-      return res.status(401).send({ error: 'Email already exists' });
+    if (usedUsername) {
```

Comment:
I suggest assigning the username to the user like we did in the 3rd party auth. That way we won't need to check the database twice for existing users.

Reply:
I didn't quite get this one. Are you suggesting auto-generated usernames, and allowing users only to choose their email addresses?

Reply:
exactly